### PR TITLE
Add Prefix & Suffix Widget in TextField

### PR DESCRIPTION
**Pull request by Aditya-Mi**

### CODE REVIEW

**Comment by Utkarsh4517** [comment]
Hey @Aditya-Mi  everything is good. Awesome work.
But currently there is no way to change the icon color as we are only taking `IconData` so I guess it would be much better if we can take `Icon` itself which will allow users to change the color of `Prefix` and `Suffix` Icons. 
Else everything LGTM 👍🏼

**Comment by Utkarsh4517** [comment]
@Aditya-Mi  also we can still see the space for icons even when no icon is provided which should not happen at all.
image

File: lib/src/input_fields/primary_input_field.dart [hunk 1]
```diff
@@ -88,14 +80,8 @@ class _MUIPrimaryInputField extends State<MUIPrimaryInputField> {
...
-          suffixIcon: IconButton(
-            onPressed: widget.suffixIconOnPressed,
-            icon: Icon(widget.suffixIcon),
-          ),
-          prefixIcon: IconButton(
-            onPressed: widget.prefixIconOnPressed,
-            icon: Icon(widget.prefixIcon),
-          ),
+          suffixIcon: widget.suffixIcon,
+          prefixIcon: widget.prefixIcon,
```
Comment (omjogani) [inline lib/src/input_fields/primary_input_field.dart:84]: Please don't let users to implement whole icon button instead what you can do...

You can stick with your previous implementation and only give IconButton if `widget.suffixIcon` is not null...

```dart
decoration: InputDecoration(
prefixIcon: widget.prefixIcon != null ? IconButton(...) : null,
suffixIcon: widget.suffixIcon != null ? IconButton(...) : null,
)
```

**Review by omjogani — CHANGES_REQUESTED** [review]
I've provided a comment please resolve it, And you should stick with your previous implementation which was better... 

Waiting for your commit and looking forward to merge!

**Pull request closed without merging**